ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/Rejections.scala

```diff
@@ -24,22 +24,24 @@ private[transaction] class Rejections(metrics: Metrics) {
 
   private final val logger = ContextualizedLogger.get(getClass)
 
-  def buildRejectionStep[A](
+  def buildImmediateRejectionStep[A](
       transactionEntry: DamlTransactionEntrySummary,
       rejection: Rejection,
       recordTime: Option[Timestamp],
-  )(implicit loggingContext: LoggingContext): StepResult[A] = {
-    buildRejectionStep(
+  )(implicit loggingContext: LoggingContext): StepResult[A] =
+    buildImmediateRejectionStep(
       buildRejectionEntry(transactionEntry, rejection),
+      rejection.description,
       recordTime,
     )
-  }
 
-  def buildRejectionStep[A](
+  def buildImmediateRejectionStep[A](
       rejectionEntry: DamlTransactionRejectionEntry.Builder,
+      rejectionDescription: String,
```

Comment:
wouldn't it make more sense for this description to be derived from the entry itself? With time there's a chance that we update only one of them and it creates very confusing logs.

Reply:
This variant is used only by the deduplication check that doesn't have a dedicated reason object. @SamirTalwar-DA is it better to add a reason for deduplication?

Reply:
Yes, but there's much larger changes to be made here to represent _all_ possible rejection reasons in a structured manner. I suggest we do it with a bit of up-front design as we need to support anything that gets written to the ledger.

Reply:
Alright, I suspected as much. I'll leave it as it is for now.